admin-dev/themes/new-theme/js/components/modal/progress-modal.ts

```diff
@@ -0,0 +1,322 @@
+}
+export type InputProgressModalParams = Partial<ProgressModalParams>;
+
+/**
+ * This class is used to build the modal DOM elements, it is not usable as is because it doesn't even have a show
+ * method and the elements are created but not added to the DOM. It just creates a basic DOM structure of a
+ * Bootstrap modal, thus keeping the logic class of the modal separated.
+ *
+ * This container is built on the basic ModalContainer and adds some confirm/cancel buttons along with a message
+ * in the body, it is mostly used as a Rich confirm dialog box.
+ */
+export class ProgressModalContainer extends ModalContainer implements ProgressModalContainerType {
+  footer!: HTMLElement;
+
+  progress!: HTMLElement;
+
+  progressInfo!: HTMLElement;
+
+  progressSuccess!: HTMLElement;
+
+  progressFailure!: HTMLElement;
+
+  progressNext!: HTMLElement;
+
+  closeButton!: HTMLElement;
+
+  successCount = 0;
+
+  failCount = 0;
+
+  /* This constructor is important to force the input type but ESLint is not happy about it*/
+  /* eslint-disable no-useless-constructor */
+  constructor(params: ProgressModalParams) {
+    super(params);
+  }
+
+  buildModalContainer(params: ProgressModalParams): void {
+    super.buildModalContainer(params);
+
+    this.buildProgress(params.modalTitle, params.modalProgressTitle, params.modalFailureTitle);
+
+    let progressDetails = document.createElement('div');
+    progressDetails.classList.add(
+      'float-right',
+      'progress-details-text',
+    );
+
+    this.body.append(progressDetails);
+
+    this.body.append(this.progress);
+
+    let errorContainer = document.createElement('div');
+    errorContainer.classList.add('modal-error-container');
+    this.body.append(errorContainer);
+    // Modal footer element
+    this.footer = document.createElement('div');
+    this.footer.classList.add('modal-footer');
+
+    // Modal close button element
+    this.closeButton = document.createElement('button');
+    this.closeButton.setAttribute('type', 'button');
+    this.closeButton.setAttribute('disabled', 'true');
+    this.closeButton.classList.add('btn', 'btn-outline-secondary', 'btn-lg');
+    this.closeButton.dataset.dismiss = 'modal';
+    this.closeButton.innerHTML = params.closeButtonLabel;
+
+    // Appending element to the modal
+    this.footer.append(this.closeButton, ...params.customButtons);
+    this.content.append(this.footer);
+  }
+
+  buildProgress(modalTitle: string, modalProgressTitle: string, modalFailureTitle: string)
+  {
+    this.buildProgressBar();
+    this.buildProgressInfo(modalTitle);
+    this.buildProgressSuccess(modalProgressTitle);
+    this.buildProgressError(modalFailureTitle);
+
+    this.progress.append(this.progressSuccess);
+    this.progress.append(this.progressFailure);
+    this.progress.append(this.progressInfo);
+  }
+
+  buildProgressBar()
+  {
+
+    this.progress = document.createElement('div');
+    this.progress.setAttribute('style', 'display: block; width: 100%');
```

Comment:
I believe you can do:
```suggestion
    this.progress.style.display = 'block';
    this.progress.style.widht = '100%';
```

but it's just a suggestion both are fine